packages/features/insights/server/virtual-queues.ts

```diff
@@ -1,81 +1,72 @@
 import { readonlyPrisma as prisma } from "@calcom/prisma";
-import { SchedulingType } from "@calcom/prisma/enums";
+import type { App_RoutingForms_Form } from "@calcom/prisma/client";
 import { getSerializableForm } from "@calcom/routing-forms/lib/getSerializableForm";
-import { zodRoutes } from "@calcom/routing-forms/zod";
 
 class VirtualQueuesInsights {
   static async getUserRelevantTeamRoutingForms({ userId }: { userId: number }) {
-    const teamRoutingForms = await prisma.app_RoutingForms_Form.findMany({
-      where: {
-        team: {
-          members: {
-            some: {
-              userId,
-            },
-          },
-        },
-      },
-      include: {
-        team: {
-          select: {
-            parentId: true,
-            parent: {
-              select: {
-                slug: true,
-              },
-            },
-            metadata: true,
-          },
-        },
-        user: {
-          select: {
-            id: true,
-            username: true,
-            movedToProfileId: true,
-          },
-        },
-      },
-    });
+    type FormWithRelations = App_RoutingForms_Form & {
+      team: {
+        parentId: number | null;
+        parent: { slug: string } | null;
+        metadata: any;
+      };
+      user: {
+        id: number;
+        username: string | null;
+        movedToProfileId: number | null;
+      };
+    };
 
+    const formsRedirectingToWeightedRR = await prisma.$queryRaw<FormWithRelations[]>`
+      WITH RECURSIVE json_array_elements_recursive AS (
+        SELECT f.id, f."teamId",
+               jsonb_array_elements(f.routes::jsonb) as route
+        FROM "App_RoutingForms_Form" f
+        WHERE f."teamId" IN (
+          SELECT "teamId"
+          FROM "Membership"
+          WHERE "userId" = ${userId}
+        )
+      )
```

Comment:
Recursive is always pretty scary but TLDR: We get all routing forms -> filter all forms to ones we are a member of -> expand routes into a single row so we can query it below using sql. 

Example: 
```json
{
  "id": 1,
  "teamId": 123,
  "routes": [
    { "action": { "eventTypeId": 456 } },
    { "action": { "eventTypeId": 789 } }
  ]
}
```

Output: 
```
id | teamId | route
1  | 123    | { "action": { "eventTypeId": 456 } }
1  | 123    | { "action": { "eventTypeId": 789 } }
```

Allows us to join later in the query on event type to get an output as follows

```
form.id | route.eventTypeId | eventType.id | eventType.schedulingType
form1   | 123              | 123          | roundRobin
form1   | 456              | 456          | collective
```